### ISSUE TITLE
typing is missing for most optimizers

### ISSUE THREAD

**Issue by pmeier** [pytorch/pytorch#40426]
## 🐛 Bug

Typing is missing for most optimizers.

## To Reproduce

1. 

 ```python
import torch
from torch import optim

params = (torch.empty(1),)

adadelta = optim.Adadelta(params)
adagrad = optim.Adagrad(params)
adam = optim.Adam(params)  # checks fine
adamax = optim.Adamax(params)
adamw = optim.AdamW(params)  # checks fine
asgd = optim.ASGD(params)
lbfgs = optim.LBFGS(params)
rmsprop = optim.RMSprop(params)
rprop = optim.Rprop(params)
sgd = optim.SGD(params, lr=1.0)  # checks fine
sparse_adam = optim.SparseAdam(params)
```

2. Run `mypy` on this.

```
6: error: Module has no attribute "Adadelta"
7: error: Module has no attribute "Adagrad"
9: error: Module has no attribute "Adamax"; maybe "Adam"?
11: error: Module has no attribute "ASGD"; maybe "SGD"?
12: error: Module has no attribute "LBFGS"
13: error: Module has no attribute "RMSprop"; maybe "Rprop"?
14: error: Module has no attribute "Rprop"; maybe "RMSprop"?
16: error: Module has no attribute "SparseAdam"
```

The only optimizers that check correctly are `Adam`, `AdamW`, and `SGD`. 

## Expected behavior

Typecheck without error.

I can't pin down why this is happening. Every optimizer has an accompanying `*.pyi` file and as far as I can see there is no difference between them. I'm especially confused by `RMSprop` and `Rprop` since `mypy` seems to recognize them and suggests one for the other. 

## Environment

```
PyTorch version: 1.5.1
Is debug build: No
CUDA used to build PyTorch: 10.2

OS: Ubuntu 18.04.4 LTS
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
CMake version: Could not collect

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 10.2.89
GPU models and configuration: GPU 0: GeForce GTX 1080
Nvidia driver version: 440.33.01
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5

Versions of relevant libraries:
[pip3] numpy==1.19.0
[pip3] numpy-stubs==0.0.1
[pip3] torch==1.5.1
[pip3] torchvision==0.6.1
[conda] Could not collect
```

cc @vincentqb @ezyang @malfet @rgommers

**Comment by yl-to** [comment]
> ## 🐛 Bug
> Typing is missing for most optimizers.
> 
> ## To Reproduce
> Steps to reproduce the behavior:
> 
> 
> ```python
> import torch
> from torch import optim
> 
> params = (torch.empty(1),)
> 
> adadelta = optim.Adadelta(params)
> adagrad = optim.Adagrad(params)
> adam = optim.Adam(params)  # checks fine
> adamax = optim.Adamax(params)
> adamw = optim.AdamW(params)  # checks fine
> asgd = optim.ASGD(params)
> lbfgs = optim.LBFGS(params)
> rmsprop = optim.RMSprop(params)
> rprop = optim.Rprop(params)
> sgd = optim.SGD(params, lr=1.0)  # checks fine
> sparse_adam = optim.SparseAdam(params)
> ```
> 
> 1. Run `mypy` on this.
> 
> ```
> 6: error: Module has no attribute "Adadelta"
> 7: error: Module has no attribute "Adagrad"
> 9: error: Module has no attribute "Adamax"; maybe "Adam"?
> 11: error: Module has no attribute "ASGD"; maybe "SGD"?
> 12: error: Module has no attribute "LBFGS"
> 13: error: Module has no attribute "RMSprop"; maybe "Rprop"?
> 14: error: Module has no attribute "Rprop"; maybe "RMSprop"?
> 16: error: Module has no attribute "SparseAdam"
> ```
> 
> The only optimizers that check correctly are `Adam`, `AdamW`, and `SGD`.
> 
> ## Expected behavior
> Typecheck without error.
> 
> I can't pin down why this is happening. Every optimizer has an accompanying `*.pyi` file and as far as I can see there is no difference between them. I'm especially confused by `RMSprop` and `Rprop` since `mypy` seems to recognize them and suggests one for the other.
> 
> ## Environment
> ```
> PyTorch version: 1.5.1
> Is debug build: No
> CUDA used to build PyTorch: 10.2
> 
> OS: Ubuntu 18.04.4 LTS
> GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
> CMake version: Could not collect
> 
> Python version: 3.6
> Is CUDA available: Yes
> CUDA runtime version: 10.2.89
> GPU models and configuration: GPU 0: GeForce GTX 1080
> Nvidia driver version: 440.33.01
> cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5
> 
> Versions of relevant libraries:
> [pip3] numpy==1.19.0
> [pip3] numpy-stubs==0.0.1
> [pip3] torch==1.5.1
> [pip3] torchvision==0.6.1
> [conda] Could not collect
> ```
> 
> cc @vincentqb @ezyang @malfet @rgommers

I tried this in the latest 1.7 Pytorch and with python 3.7 installed, mypy shows:
`Success: no issues found in 1 source file`
Maybe this problem was fixed in newer versions?

**Comment by ezyang** [comment]
Yes, it was fixed in 159835e66669b593ee14eff895d0829060adcdb4

**Comment by pmeier** [comment]
@ezyang Indeed this is fixed in the current nightly, but not from #31130 alone. The stubs landed with `torch=1.5.0` and I'm seeing these issues in `torch==1.5.1`. I suspect #36358 is the missing part in the current stable release.

**Comment by ezyang** [comment]
That may very well be true. 1.6.0 is coming soon so it's unlikely there will be a 1.5.2 release to fix this issue.
